libs/common/src/main/java/org/opensearch/common/round/RoundableFactory.java

```diff
@@ -29,7 +29,7 @@ private RoundableFactory() {}
     /**
      * Creates and returns the fastest implementation of {@link Roundable}.
      */
-    public static Roundable create(long[] values, int size) {
+    public static Roundable create(long[] values, int size, boolean useSimdIfAvailable) {
```

Comment:
@ketanv3 I think the large part of the complexity with this pull request is related to the fact that you are trying to lift the SIMD part up the API level. I think that:
1) The `RoundFactory` API should get rid of `useSimdIfAvailable` flag
2) The `RoundFactory` for JDK-20 should check the feature flag internally (`create` method) and make the appropriate decision

Reply:
That's what I wanted to do in the first place, only to realize later that [FeatureFlags](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/util/FeatureFlags.java) (and equivalently Settings) are defined in the server module.

Would have been cleaner if the settings' framework was part of some common module, but we aren't there yet. Let me know if you have a better alternative.

Reply:
Ah that is true, I missed that part. The feature flag is not really applicable here (as the concept) but the system property is. I think we could remove the notion of feature flag in favour of system property.

Reply:
(why it is not applicable - the `opensearch-common` is essentially a standalone module which could be used outside of OpenSearch whereas feature flags are very specific to OpenSearch).

Reply:
I wasn't sure about using system properties over OpenSearch settings. But I'm totally with you on keeping this module independent of OpenSearch.

I'll update it to load the feature flag from the system properties directly. Thanks!

Reply:
> I'll update it to load the feature flag from the system properties directly. Thanks!

Thanks, yeah, we use system properties in a few places (where  feature flags are not applicable)

Reply:
I've made the suggested change. I'm also inclined to remove the "opensearch" prefix from the system property name "opensearch.experimental.feature.simd.rounding.enabled" since this module doesn't know anything about OpenSearch. Thoughts?